src/Uri.php

```diff
@@ -269,13 +280,16 @@ private static function createUriString(string $scheme, string $authority, strin
      */
     private static function isNonStandardPort(string $scheme, int $port): bool
     {
-        return !isset(self::$schemes[$scheme]) || $port !== self::$schemes[$scheme];
+        $scheme_not_set = !isset(self::$schemes[$scheme]);
+        $is_invalid_port = $port !== self::$schemes[$scheme];
```

Comment:
This is the line that breaks the code. Our tests correctly catch this, thus Travis fails.

Your two lines of code are not the same as the old one line of code. For one line of code, the second part of the “or” statement does not run when the first part evaluates as false. We specifically make use of this by checking if our array contains an item with the key `$scheme` before we access it.

Your changed code is always going to try and access the array item with the key `$scheme`. When you ask PHP to do that and there is no such item in the array, it will fail.

I think – though this is without testing or guarantees – the actual equivalent to the old code would be:

```php
if (false === isset(self::$schemes[$scheme])) {
    return true;
} else if ($port !== self::$schemes[$scheme]) {
    return true;
}
return false;
```

This means the second check will only be done when the first one didn’t pass.

Reply:
Thanks man. I rolled back that method to the initial state. So Travis check passed successfully.